src/cut_detector/factories/mb_support/tracking/spatial_laptrack.py

```diff
@@ -0,0 +1,307 @@
+""" A modified version of LapTrack.
+- Maximum distance is only applied to the euclidian distance
+"""
+
+import networkx as nx
+import numpy as np
+
+from typing import Callable, Literal, List, cast
+from pydantic import Field
+from scipy.spatial.distance import cdist
+from laptrack import LapTrack, ParallelBackend
+from functools import partial
+
+from laptrack._typing_utils import NumArray, Int, EdgeType
+from laptrack._coo_matrix_builder import coo_matrix_builder
+from laptrack._optimization import lap_optimization
+from laptrack._cost_matrix import build_frame_cost_matrix
+
+class SpatialLapTrack(LapTrack):
+    """
+    SpatialLapTrack is a modification of LapTrack where
+    track_cost_cutoff and gap_closing_cost_cutoff are compared
+    against euclidian spatial distance only.
+
+    (splitting and merging cost cutoff works as usual because
+    they are not used in the project for now)
+
+    This is only for filtering out values: 
+    Optimisation is still run against the chosen metric.
+    """
+
+    spatial_coord_slice: slice = Field(
+        ...,
+        description="A slice that is used to subset coords"
+    )
+
+    spatial_metric: str | Callable  = Field(
```

Comment:
Ce truc crée une erreur : 
TypeError: unsupported operand type(s) for |: 'type' and '_CallableType', et les tests passent plus...